lib/src/grid/grid.component.ts

```diff
@@ -70,13 +81,42 @@ export class GridComponent implements OnChanges, AfterContentInit {
         }
     }
 
+    private masterCheckboxClicked(masterCheckboxState: boolean) {
+        this.rows.map(r => r.selected = masterCheckboxState);
+        this.fullDataSet.map(r => r.selected = masterCheckboxState);
+
+    }
+
+    private selectRow(selectEvent: SelectEvent) {
+        let row = this.fullDataSet.find(r => r === selectEvent.row);
+        let displayRow = this.rows.find(r => r === selectEvent.row);
+        row.selected = selectEvent.selected;
+        displayRow.selected = selectEvent.selected;
+        let index = this.selectedRows.findIndex(r => r === row);
+        if (index > -1) {
+            this.selectedRows.splice(index, 1);
+        } else {
+            this.selectedRows.push(row);
+        }
+
+        if (this.selectedRows.length > 0 && this.selectedRows.length < this.fullDataSet.length) {
+            this.masterCheckbox.setIndeterminateState();
+        } else if (this.selectedRows.length === 0) {
+            this.masterCheckbox.setUncheckedState();
+        } else {
+            this.masterCheckbox.setCheckedState();
+        }
+    }
+
     private sort(sortEvent: SortEvent) {
```

Comment:
many of these data manipulation/retrieving functions should be separated into a service. So they could be replaced with a call to the server. I would also consider making them work properly with observables.

Reply:
@benjanderson please do! That's why I created the separate table branch on the main fork so that other people could jump in.

Reply:
@benjanderson I think I'd prefer a sort output and a paging output to a service. I think that calls to  the server are client specific and wouldn't want to be presumptuous about how data should be retrieved. Thoughts?